### ISSUE TITLE
Update on this plugin?

### ISSUE THREAD

**Issue by ralbright-wa** [foosel/OctoPrint-DisplayZ#10]
I installed this plugin hoping to just be able to see what layer Octoprint had recently sent to the printer. (Like as it is sending the command to raise layer height, add that to the variable for current layer and display that.)
I understand it would only work with files that are printed from Octoprint and not from SD card (since setting up Octoprint, I have not printed anything directly from the SD card).
I just installed this last night and sent something to one of my printers. Prior to accessing the GCode Viewer (which fails on larger prints) The current height was showing as -1.00mm. When I access GCode viewer, that showed layer number at 312 and for a moment the current height did update to show the correct height, of 31.4mm, but then jumped back to -1.00mm.

Is what I am suggesting even possible, just to start processing passed on Gcode and intercept the commands sent to the printer, and update the variable for height based on a few commands? When you see a z-layer increase to increase the current height, and when you see a decrease decrease the current height? (This may produce false positives due to retractions which is why I state show the current print head's position based on any z command.)

**Comment by Sunoo** [comment]
I’ve not seen it display -1 ever, it’s continuing to work as expected with both of my printers as of the most recent prints I did a week or so ago. Is there something unique or unusual with your setup or gcode file? Have you seen it with every print, or have you only tried it the one time so far?

**Comment by ralbright-wa** [comment]
First print with it so far. Nothing unique with the printer. It is an Anycubic Mega X. When the G-Code viewer is not active (say on temperature screen it shows -1.00mm, but if I go to the G-Code viewer after about 5-10 seconds it updates to show the current height same as in G-Code viewer.

**Comment by Sunoo** [comment]
Interesting, any chance you could share the gcode file? You could email it to me if it’s not something you want to have be public.

**Comment by ralbright-wa** [comment]
Yes I can share the gcode, email sent to your profile email address.

**Comment by Sunoo** [comment]
Great, I’ll try running that through Octoprint this weekend and see if I can recreate the issue.

**Comment by ralbright-wa** [comment]
A few more prints and added it to my other printers. It is appearing to work better, but at times still drops to -1.00mm. Not sure what that is about though
